src/main/java/net/spy/memcached/MemcachedClient.java

```diff
@@ -226,9 +226,9 @@ public MemcachedClient(List<InetSocketAddress> addrs) throws IOException {
   }
 
   public MemcachedClient(ConnectionFactory cf, List<InetSocketAddress> addrs) throws IOException{
-    this(cf, addrs, false);
+    this(cf, addrs, cf != null && cf.getClientMode() == ClientMode.Unset);
```

Comment:
If the ConnectionFactory object is `null`, then we should discover the mode for the client.

Reply:
`(cf != null && cf.getClientMode() == ClientMode.Unset) || cf ==null` should be able to cover it.